x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/DateScriptFieldScript.java

```diff
@@ -44,31 +45,90 @@ public DateScriptFieldScript(Map<String, Object> params, SearchLookup searchLook
         this.formatter = formatter;
     }
 
-    public static class Millis {
-        private final DateScriptFieldScript script;
+    public static long[] convertFromLong(long v) {
+        return new long[] { v };
+    }
 
-        public Millis(DateScriptFieldScript script) {
-            this.script = script;
-        }
+    public static long[] convertFromTemporalAccessor(TemporalAccessor v) {
+        return new long[] { convertSingleTemporalAccessor(v) };
+    }
 
-        public void millis(long v) {
-            script.collectValue(v);
+    public static long[] convertFromTemporalAccessorArray(TemporalAccessor[] v) {
+        long[] result = new long[v.length];
+        int i = 0;
+        for (TemporalAccessor o : v) {
+            result[i++] = convertSingleTemporalAccessor(o);
         }
+        return result;
     }
 
-    public static class Date {
-        private final DateScriptFieldScript script;
+    public static long[] convertFromCollection(Collection<?> v) {
+        long[] result = new long[v.size()];
+        int i = 0;
+        for (Object o : v) {
+            try {
+                result[i++] = convertCollectionElement(o);
+            } catch (ClassCastException e) {
+                throw new ClassCastException("Exception casting collection member [" + o + "]: " + e.getMessage());
+            }
+        }
+        return result;
+    }
 
-        public Date(DateScriptFieldScript script) {
-            this.script = script;
+    public static long[] convertFromDef(Object o) {
```

Comment:
This one should be temporary, waiting on improvements in painless. But I think the improvement in painless will amount to them generating the same `instanceof` tree.